pkg/pod/ttlDestroyHandler.go

```diff
@@ -32,16 +35,19 @@ func (t ttlDestroyHandler) Supports() string {
 }
 
 func (t ttlDestroyHandler) Handle(pod v1.Pod) error {
-	log.Printf("TTL handler invoked: %s", pod.Name)
+	correlationID := uuid.NewString()
+	t.logger = t.logger.With("correlationId", correlationID)
+
+	t.logger.Info("TTL handler invoked", "pod", pod.Name)
 
 	creationTimestampLabel := pod.Labels["im-creation-timestamp"]
 	if creationTimestampLabel == "" {
-		return errors.New("no creationTimestamp label found")
+		return fmt.Errorf("failed to find label \"im-creation-timestamp\" on pod %q", pod.Name)
 	}
 
 	ttlLabel := pod.Labels["im-ttl"]
 	if ttlLabel == "" {
-		log.Println("no TTL label found")
+		t.logger.Info(`No TTL label "im-ttl" found`)
```

Comment:
Are you not using "Failed to find ..." prefix here becuse the log level is INFO?